airbyte-server/src/main/java/io/airbyte/server/ConfigurationApiFactory.java

```diff
@@ -53,10 +55,15 @@ public static void setAirbyteVersion(final String airbyteVersion) {
     ConfigurationApiFactory.airbyteVersion = airbyteVersion;
   }
 
+  public static void setDatabase(Database database) {
```

Comment:
we can leave this as for this PR and do a refactor after... but instead of passing airbyte version and the database instead we should add a new class called `AirbyteArchiver` that takes in the airbyte version and database in its constructor. it can also contain the DatabaseArchiver and the ConfigArchiver. Then the ArchiveHandler would just called createArchive and extractArchive (or equivalent).

there are 2 things triggering this:
1. in general i try to keep *Handler classes really stupid where possible. This helps force separation of the API interface and anything complex that needs to be tested. It would also be nice to encapsulate all of this archiving logic in one place.
2. i want to avoid passing around the raw database.
3. we now have multiple attributes (yeah yeah, only 2) that we are passing through a bunch of different classes just to construct the ArchiveHandler.

Reply:
the `schedulerJobClient` is a good example of what it would look like.

Reply:
If we use the JobPersistence in archive handler, we only need to pass 1 extra argument (the version)